java/libs/src/main/java/org/physical_web/collection/PhysicalWebCollection.java

```diff
@@ -245,6 +245,39 @@ public void jsonDeserialize(JSONObject jsonObject) throws PhysicalWebCollectionE
    */
   public List<PwPair> getPwPairsSortedByRank() {
     // Get all valid PwPairs.
+    List<PwPair> allPwPairs = getPwPairs();
```

Comment:
Maybe I'm late here but why returning a list of PwPair and not a simply list or PwResult.
The PwResult could embed the UrlDevice or could include at least the initial URL of the UrlDevice.
That would be one less datatype to deal with from the API perspective.